src/Fhir.Metrics/Utils/Parser.cs

```diff
@@ -51,17 +51,41 @@ public bool IsNumeric
     public static class Parser
     {
 
-        public static string TokenPattern = @"(((?<m>[\.\/])?(?<m>[^\.\/]+))*)?";
+        public static Regex TokenPattern = new Regex(@"(((?<m>[\.\/])?(?<m>[^\.\/]+))*)?", RegexOptions.Compiled);
+        public static Regex Annotations = new Regex(@"{[^{}]*}", RegexOptions.Compiled);
 
         public static List<string> Tokenize(string expression)
         {
-            bool valid = Regex.IsMatch(expression, TokenPattern);
-            if (!valid)
+            if(Annotations.IsMatch(expression))
+                expression = CanonicalizeAnnotations(expression);
+
+            if (!TokenPattern.IsMatch(expression) || Annotations.IsMatch(expression))
                 throw new ArgumentException("Invalid metric expression");
 
-            Match match = Regex.Match(expression, TokenPattern);
-            return match.Captures("m").ToList();
+            return TokenPattern.Match(expression).Captures("m").ToList();
```

Comment:
This code was already inefficient in that it did an IsMatch and then a Match. Both Annotations and TokenPattern benefit from a refactor to:
``` csharp
  var match = TokenPattern.Match(expression);
  if (match.Success) ...
```